File: VendingMachine.Core/Models/Coin.cs
namespace VendingMachine.Core.Models
{
    // coin and price
    public class Coin
    {
        public TypeCoin TypeCoin { get; set; }

        public Coin(TypeCoin typeCoin)
        {
            TypeCoin = typeCoin;
        }

        // цена монеты в int
        public int Price => (int)this.TypeCoin;
    }
}

File: VendingMachine.UI.Angular/src/app/models/coin.ts
import { TypeCoin } from './enums';

export class Coin {
  typeCoin: TypeCoin;

  public price = () => {
    return TypeCoin[this.typeCoin];
  }

  constructor(typeCoin: TypeCoin) {
    this.typeCoin = typeCoin;
  }
}
